likecoin/admin/restful.php

```diff
@@ -45,6 +45,167 @@ function likecoin_rest_refresh_publish_status( $request ) {
+	foreach ( $urls as $url ) {
+		$file_info  = new finfo( FILEINFO_MIME_TYPE );
+		$image_path = substr( $url, 22 );
+		// phpcs:disable WordPress.WP.AlternativeFunctions
+		$img_body = file_get_contents( $image_path );
+		// phpcs:enable WordPress.WP.AlternativeFunctions
+		$mime_type = $file_info->buffer( $img_body );
+		$filename  = basename( $url );
+		$body     .= '--' . $boundary . "\r\n";
+		$body     .= 'Content-Disposition: form-data; name="' . $filename . '"; filename="' . $filename . "\"\r\n";
+		$body     .= 'Content-Type: ' . $mime_type . "\r\n";
+		$body     .= "Content-Transfer-Encoding: binary\r\n";
+		$body     .= "\r\n";
+		$body     .= $img_body . "\r\n";
+		$body     .= "\r\n";
+	}
+	$body .= '--' . $boundary . '--';
+	return $body;
+}
+/**
+ * Add likecoin arweave upload endpoint.
+ *
+ * @param WP_REST_Request $request Full data about the request.
+ * @return WP_Error|WP_REST_Response
+ */
+function likecoin_rest_post_arweave_upload( $request ) {
+	$post_id = $request['id'];
+	$post    = get_post( $post_id );
+	$tx_hash = $request['txHash'];
+	if ( ! isset( $post ) ) {
+		return new WP_Error( 'post_not_found', __( 'Post was not found', LC_PLUGIN_SLUG ), array( 'status' => 404 ) );
+	}
+	// phpcs:disable WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+	$boundary = base64_encode( wp_generate_password( 24 ) );
+	// phpcs:enable WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+	$body                    = likecoin_format_post_to_multipart_formdata( $boundary, $post );
+	$likecoin_api_upload_url = 'https://api.like.co/api/arweave/upload?txHash=' . $tx_hash; // TODO: change based on test/main net.
+	$response                = wp_remote_post(
+		$likecoin_api_upload_url,
+		array(
+			'headers' => array(
+				'Content-Type' => 'multipart/form-data; boundary=' . $boundary,
+			),
+			'body'    => $body,
+		)
+	);
+	if ( is_wp_error( $response ) ) {
+		$err_message = $response->get_error_message();
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			// phpcs:disable WordPress.PHP.DevelopmentFunctions
+			error_log( $response->get_error_message() );
+			// phpcs:enable WordPress.PHP.DevelopmentFunctions
+		}
+		return array( 'error' => $response->get_error_message() );
+	}
+	$decoded_response = json_decode( $response['body'], true );
+	if ( ! $decoded_response ) {
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			// phpcs:disable WordPress.PHP.DevelopmentFunctions
+			error_log( $response['body'] );
+			// phpcs:enable WordPress.PHP.DevelopmentFunctions
+		}
+		return array( 'error' => $response['body'] );
+	}
+	// save arweaveId & ipfsHash to WordPress DB.
+	$arweave_info = get_post_meta( $post_id, LC_ARWEAVE_INFO, true );
+	if ( ! is_array( $arweave_info ) ) {
+		$arweave_info = array();
+	}
+	$arweave_info['arweave_id'] = $decoded_response['arweaveId'];
+	$arweave_info['ipfs_hash']  = $decoded_response['ipfsHash'];
+	update_post_meta( $post_id, LC_ARWEAVE_INFO, $arweave_info );
```

Comment:
I think it is unexpected for a `restful` call to also have a side effect that calls `update_post_meta`
Better use a wrapper function that calls `likecoin_rest_post_arweave_upload` , and save the result into post meta in that function